packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneBlobs.spec.ts

```diff
@@ -0,0 +1,192 @@
+
+        // Reference a blob - important for making it live
+        defaultDataObject._root.set(handleKey, blobHandle);
+
+        // Unreference a blob
+        defaultDataObject._root.delete(handleKey);
+
+        // Summarize
+        const {
+            container: summarizingContainer1,
+            summarizer: summarizer1,
+        } = await loadSummarizerAndContainer();
+        const summaryVersion = (await summarize(summarizer1)).summaryVersion;
+
+        // Close the containers as these containers would be closed by session expiry before sweep ready ever occurs
+        container.close();
+        summarizingContainer1.close();
+
+        // Wait some time, the datastore can be in many different unreference states
+        await delay(approximateUnreferenceTimestampMs);
+
+        // Load a new container and summarizer based on the latest summary, summarize
+        const {
+            container: summarizingContainer2,
+            summarizer: summarizer2,
+        } = await loadSummarizerAndContainer(summaryVersion);
+
+        return {
+            absolutePath: blobHandle.absolutePath,
+            summarizingContainer: summarizingContainer2,
+            summarizer: summarizer2,
+            summaryVersion,
+        };
+    };
+
+    // If this test starts failing due to runtime is closed errors try first adjusting `sweepTimeoutMs` above
+    itExpects("Handle request for tombstoned blobs fails in summarizing container loaded after sweep timeout",
+    [
+        {
+            error: "GC_Tombstone_Blob_Requested",
+            eventName: "fluid:telemetry:BlobManager:GC_Tombstone_Blob_Requested",
+            viaHandle: true,
+        },
+    ],
+    async () => {
+        const {
+            absolutePath,
+            summarizingContainer,
+            summarizer,
+        } = await summarizationWithUnreferencedBlobAfterTime(sweepTimeoutMs);
+        // The blob should be tombstoned now
+        const { summaryVersion } = await summarize(summarizer);
+
+        const container = await loadContainer(summaryVersion);
+        const defaultDataObject = await requestFluidObject<ITestDataObject>(container, "default");
+        const fluidHandleContext = defaultDataObject._context.containerRuntime.IFluidHandleContext;
+
+        // Handle requests for blob handle should fail!
+        const response = await fluidHandleContext.resolveHandle({ url: absolutePath });
+        assert(response?.status === 404, `Expecting a 404 response!`);
+        assert(response.value.startsWith("Blob removed by gc:"));
+        assert(summarizingContainer.closed !== true, "Summarizing container should not have closed!");
+        assert(container.closed !== true, "Container does not close");
+    });
+
+    // If this test starts failing due to runtime is closed errors try first adjusting `sweepTimeoutMs` above
+    itExpects("Handle request for tombstoned blobs only logs in summarizing container loaded after sweep timeout",
```

Comment:
This test is also testing non-summarizer container. Also, its testing that with the ThrowOnTombstoneUsage set to false, there is no error thrown.

Reply:
The summarizing container would close if we ThrowOnTombstoneUsage. So it's indirectly tested here.

Reply:
Sure but you should fix the test names to reflect what they are testing. Having an explicit test for summarizer container would be useful too. Also, summarizer container should not close (you said would).

Reply:
I'll add an extra test here, I originally thought adding another test would be redundant since it's already indirectly tested.